mindocr/optim/adan.py

```diff
@@ -163,6 +163,8 @@ def construct(self, gradients):
         gradients = self.scale_grad(gradients)
         gradients = self._grad_sparse_indices_deduplicate(gradients)
         lr = self.get_lr()
+        if version.__version__ >= "2.2" and self._is_dynamic_lr_or_weight_decay():
```

Comment:
`self._is_dynamic_lr_or_weight_decay()` this condition is not necessary. No matter dynamic lr or constant lr, we should increase global_step in each optimization step.

Reply:
The PR has been revised according to this review comments. The condition `self._is_dynamic_lr_or_weight_decay()` is unnecessary because `self.global_step` is only used for `is_dynamic_lr` and `is_weight_decay` scenarios.